gui/functionality/add_cluster.py

```diff
@@ -37,12 +37,28 @@ def __init__(self, data_provider, parent=None):
         self.ui.key_type_box.setCurrentText(self.parameters['add_cluster_arguments']['key_type'])
         self.ui.add_actions_file_button.clicked.connect(self.open_actions_file_dialog)
         self.ui.delete_actions_file_button.clicked.connect(self.delete_actions_file)
+        self.ui.auth_method_box.currentIndexChanged.connect(self.auth_method_clicked)
+        self.ui.add_key_button.clicked.connect(self.add_key)
+        self.ui.delete_key_button.clicked.connect(self.delete_key)
 
         self.actions_file_name = None
+        self.key_name = None
 
         lay = QVBoxLayout(self)
         lay.addWidget(self.ui)
 
+    def auth_method_clicked(self):
```

Comment:
I suggest `selected` or `changed` instead of `clicked` as the name of the function.